NetEase/arctic
# [Arctic-1160][Spark]: Unit test refactor for spark module

## Why are the changes needed?

To fix #1160 


## Brief change log
This PR is in progress and the targets are:

- Upgrade junit version from 4.x to 5.x
- Migrate test code to new framework
- Add a spark-3-common module to reuse spark sql tests.
- The code cover of spark tests will be imporved, the core class and core process will be improved to 90%.

## How was this patch tested?
- Add some test cases that check the changes thoroughly including negative and positive cases if possible
- Run test locally before making a pull request
